src/tensors/levels/abstractlevel.jl

```diff
@@ -0,0 +1,18 @@
+abstract type AbstractLevel end
+abstract type AbstractVirtualLevel end
+
+#is_laminable_updater(lvl::AbstractVirtualLevel, ctx, ::Union{::typeof(defaultread), ::typeof(walk), ::typeof(gallop), ::typeof(follow), typeof(defaultupdate), typeof(laminate), typeof(extrude)}, protos...) = false
+
+#is_laminable_updater(lvl::AbstractVirtualLevel, ctx) = false
+
+
+
+is_concurrent(lvl::AbstractVirtualLevel, ctx) = true
+
+is_injective(lvl::AbstractVirtualLevel, ctx, accs) = false
+
+
+getroot(tns::AbstractVirtualLevel) = tns
```

Comment:
getroot should not be defined on a level, instead define on a fiber (which would be an `AbstractVirtualArray`